### PR TITLE
chore(api): Add stress tests for API category

### PR DESCRIPTION
**Pull request by thisisabhash**
*Issue #, if available:* None

*Description of changes:* Add stress tests for API category

*Check points: (check or cross out if not relevant)*

- [x] Added new tests to cover change, if needed
- [x] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [x] All integration tests pass
- [x] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] Documentation update for the change if required
- [x] PR title conforms to conventional commit style
- [ ] If breaking change, documentation/changelog update with migration instructions

*DataStore checkpoints (check when completed)*

- [ ] Ran AWSDataStorePluginIntegrationTests
- [ ] Ran AWSDataStorePluginV2Tests
- [ ] Ran AWSDataStorePluginMultiAuthTests
- [ ] Ran AWSDataStorePluginCPKTests
- [ ] Ran AWSDataStorePluginAuthCognitoTests
- [ ] Ran AWSDataStorePluginAuthIAMTests

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/API/Tests/APIHostApp/GraphQLAPIStressTests/README.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+
+Run `amplify codegen models` from your amplify app, compare and replace if needed the ones in the Models folder.
+
+
+### Set-up
+
+1. `amplify init`. Update `cli.json` values to use Transformer V1.
+
+- `"useexperimentalpipelinedtransformer": false`
+- `"transformerversion": 1`
```
Comment (lawmicha) [inline AmplifyPlugins/API/Tests/APIHostApp/GraphQLAPIStressTests/README.md:15]: we shouldn't use TransformerV1 as we would like to deprecate this usage, so we should especially dogfood the latest transformer process in our integration tests to make sure we're using the same backend as new developers would

File: AmplifyPlugins/API/Tests/APIHostApp/GraphQLAPIStressTests/GraphQLAPIStressTests.swift [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+    /// - When: I update 50 post simultaneously
+    /// - Then: Operation should succeed
+    func testMultipleUpdateMutations() async throws {
+        let postCreateExpectation = asyncExpectation(description: "Post was created successfully",
+                                                     expectedFulfillmentCount: concurrencyLimit)
+        let postUpdateExpectation = asyncExpectation(description: "Post was updated successfully",
+                                                     expectedFulfillmentCount: concurrencyLimit)
+        let postOffice = PostOffice()
+        
+        for index in 0..<concurrencyLimit {
```
Comment (lawmicha) [inline AmplifyPlugins/API/Tests/APIHostApp/GraphQLAPIStressTests/GraphQLAPIStressTests.swift:181]: Coming from reviewing https://github.com/aws-amplify/amplify-swift/pull/2639/files what's the difference between the different ways to simultaneously ways to update the post? In the DS PR we're using DispatchQueue.concurrentPerform, in this PR we're creating Tasks on each for loop iteration. And then there's the above test that uses a TaskGroup. Which one should we be using for stress testing?

File: AmplifyPlugins/API/Tests/APIHostApp/GraphQLAPIStressTests/GraphQLAPIStressTests.swift [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+                XCTAssertEqual(post.id, queriedPost.id)
+                XCTAssertEqual(post.title, queriedPost.title)
+                await postQueryExpectation.fulfill()
+            }
+        }
+        
+        await waitForExpectations([postQueryExpectation], timeout: TestCommonConstants.networkTimeout)
+    }
+    
+    actor PostOffice {
```
Comment (lawmicha) [inline AmplifyPlugins/API/Tests/APIHostApp/GraphQLAPIStressTests/GraphQLAPIStressTests.swift:297]: just curious, why is this called `PostOffice`?

Comment (thisisabhash) [reply AmplifyPlugins/API/Tests/APIHostApp/GraphQLAPIStressTests/GraphQLAPIStressTests.swift]: Renamed

File: AmplifyPlugins/API/Tests/APIHostApp/GraphQLAPIStressTests/README.md (hunk 1 above)
Comment (thisisabhash) [reply AmplifyPlugins/API/Tests/APIHostApp/GraphQLAPIStressTests/README.md]: Updated code

File: AmplifyPlugins/API/Tests/APIHostApp/GraphQLAPIStressTests/GraphQLAPIStressTests.swift (hunk 2 above)
Comment (thisisabhash) [reply AmplifyPlugins/API/Tests/APIHostApp/GraphQLAPIStressTests/GraphQLAPIStressTests.swift]: `TaskGroup` is used for concurrency for e.g. if there are multiple parts of your code which `async` and related, they can be put within a task group. `.concurrentPerform` is used when there are multiple instances of `async` and unrelated code. In this case, we are running multiple instances of same code on different data. Updated it to use `.concurrentPerform`

**Review by lawmicha — DISMISSED** [review]

**Review by lawmicha — APPROVED** [review]

**Pull request merged**